scripts/benchmark_ipa.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Generate stage-wise report after running IPA bench
+#
+# To use, supply the number of rows you want to run benchmarking on
+
+num_rows=$1
+if [ "$#" -ne 1 ]; then
+    echo "Please supply query size as an argument to the script"
+    exit
+fi
+
+temp_file=".benchmark_result$num_rows.csv"
+RUST_LOG=ipa=DEBUG cargo bench --bench oneshot_ipa --features="enable-benches" --no-default-features -- -n $num_rows > $temp_file
```

Comment:
There is an alternative approach that doesn't involve a temporary file.  Try:

```sh
declare -A bytes
declare -A records
RUST_LOG=ipa... cargo bench ... | while IFS=, read line k a b c; do
    # Maybe permute k so that you can get the step more easily:
    case "$k" in
      */mod_conv_breakdown_key/*|*/mod_conv_match_key/*) step=Verify ;;
      # etc...
      *) step=Other ;;
    esac
    records[$step]=$(($records[$step] + $a))
    bytes[$step]=$((${bytes[$step]} + $b))
done

for s in "${!records[@]}"; do # Or you can just enumerate the steps so that the order is consistent.
    echo "$s: ${records[$s]} records, ${bytes[$s]} bytes"
done
```

Reply:
I only used associative arrays because that is easier to manage.  You can use different variables instead; it just looks a bit ugly.

Also, blame Apple.  They refuse to ship a GPL-3 version of bash, so they are stuck in the past.

Reply:
I am having hard time to make it work. Looks like piping data makes the variable disappear since piping creates a SubShell which runs in its own environment : http://mywiki.wooledge.org/BashFAQ/024

Reply:
Took me a while to figure this out.. but hopefully it looks good now :)